packages/remark-parse/types/index.d.ts

```diff
@@ -0,0 +1,52 @@
+// TypeScript Version: 3.0
+
+import {Node, Parent, Position} from 'unist'
+import {Parser, Attacher} from 'unified'
+
+declare class RemarkParser extends Parser {
+  blockMethods: string[]
+  inlineTokenizers: {
+    [key: string]: remarkParse.Tokenizer
+  }
+  inlineMethods: string[]
+}
+
+declare namespace remarkParse {
+  interface Parse extends Attacher {
+    (options: RemarkParseOptions): void
```

Comment:
I have a test for the most common usage pattern here, which supports partial options
https://github.com/remarkjs/remark/pull/383/files#diff-e4f34330f5a7ed229a6b01775e638a1fR10. This is because a top type (`unknown`) is used here 

https://github.com/unifiedjs/unified/blob/38d2c8211e240a64891aea871caae376428d0200/types/index.d.ts#L21

Are there other usage patterns we aim to support? Is there a way to directly construct a `Parse` plugin, or is
```ts
unified().use(remarkParse, parseOptions)
```
the only way?

Reply:
> Unified types would highly benefit from a generic here. As written, options passed along with the use() call is a place where all type safety is lost.

I opened a PR with an initial implementation for what generic plugin settings could look like over at https://github.com/unifiedjs/unified/pull/58

Feel free to chime in or refactor.

Reply:
> Should this be Partial<RemarkParseOptions>?

Makes sense a lot. But rather than wrap it with `Partial`, I think we should make all props of the interface optional. @ChristianMurphy

Reply:
@mike-north unifiedjs/unified#58 has been merged and will be included in the next release.
Unified now has generics for plugins to declare their settings `interface`(s).

Reply:
> Are there other usage patterns we aim to support? Is there a way to directly construct a Parse plugin, or is
> ```js
> unified().use(remarkParse, parseOptions)
> ```
> the only way?

As far as I know, `use` is the only way official to attach `remark-parse` to Unified.
@wooorm are there any others?

Reply:
Generic has been applied